homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,304 @@
+"""EnaSolar solar inverter interface."""
+from __future__ import annotations
+
+# from datetime import time
+from datetime import date, datetime
+import logging
+
+import pyenasolar
+import voluptuous as vol
+
+from homeassistant.components.sensor import (
+    PLATFORM_SCHEMA,
+    STATE_CLASS_MEASUREMENT,
+    SensorEntity,
+)
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_NAME,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    EVENT_HOMEASSISTANT_START,
+    EVENT_HOMEASSISTANT_STOP,
+    POWER_WATT,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import CALLBACK_TYPE, callback
+from homeassistant.exceptions import PlatformNotReady
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_call_later
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_MAX_OUTPUT,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    ENASOLAR_UNIT_MAPPINGS,
+    SCAN_DATA_MIN_INTERVAL,
+    SCAN_MAX_INTERVAL,
+    SCAN_METERS_MIN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Required(CONF_SUN_DOWN, default="22:00"): cv.time,
+        vol.Required(CONF_SUN_UP, default="5:00"): cv.time,
+        vol.Optional(CONF_MAX_OUTPUT): vol.In([1.5, 2.0, 3.0, 3.8, 4.0, 5.0]),
+        vol.Optional(CONF_NAME): cv.string,
+        vol.Optional(CONF_DC_STRINGS): vol.In([1, 2]),
+        vol.Optional(CONF_CAPABILITY): cv.positive_int,
+    }
+)
```

Comment:
For integrations that connect to devices or services, we no longer accept new YAML configuration.

This integration needs to be refactored first to use a configuration flow and config entries first.

More information about this can be found in Architecture Decision Record:

- ADR-0010: <https://github.com/home-assistant/architecture/blob/master/adr/0010-integration-configuration.md#decision>

See our developer documentation on how to get started creating a configuration flow for this integration:

<https://developers.home-assistant.io/docs/config_entries_config_flow_handler>

Reply:
I believe I have re-implemented this integration using the config_flow method